symbolic-debuginfo/src/plist.rs

```diff
@@ -0,0 +1,155 @@
+//! Apple PropertyList support.
+//!
+//! For MachO objects
+
+use std::collections::HashMap;
+use std::error::Error as StdError;
+use std::fmt;
+use std::io::Cursor;
+
+use elementtree::Element;
+use thiserror::Error;
+
+use symbolic_common::{DebugId, ParseDebugIdError};
+
+/// The error type for handling a [`PList`].
+#[derive(Debug, Error)]
+#[error("{kind}")]
+pub struct PListError {
+    kind: PListErrorKind,
+    #[source]
+    source: Option<Box<dyn StdError + Send + Sync + 'static>>,
+}
+
+impl From<elementtree::Error> for PListError {
+    fn from(source: elementtree::Error) -> Self {
+        Self {
+            kind: PListErrorKind::Xml,
+            source: Some(Box::new(source)),
+        }
+    }
+}
+
+impl From<ParseDebugIdError> for PListError {
+    fn from(source: ParseDebugIdError) -> Self {
+        Self {
+            kind: PListErrorKind::Parse,
+            source: Some(Box::new(source)),
+        }
+    }
+}
+
+/// Error kind for [`PListError`].
+#[non_exhaustive]
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum PListErrorKind {
+    /// The plist did not have the expected XML schema.
+    Schema,
+    /// There was an XML parsing error.
+    Xml,
+    /// Failed to parse a PList value.
+    Parse,
+}
+
+impl fmt::Display for PListErrorKind {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Self::Schema => write!(f, "XML structure did not match expected schema"),
+            Self::Xml => write!(f, "Invalid XML"),
+            Self::Parse => write!(f, "Failed to parse a value into the right type"),
+        }
+    }
+}
+
+impl From<PListErrorKind> for PListError {
+    fn from(source: PListErrorKind) -> Self {
+        Self {
+            kind: source,
+            source: None,
+        }
+    }
+}
+
+/// Apple PropertyList structure.
+///
+/// This is an auxiliary file, not an object file.
+///
+/// The PList format is used to map the UUID of a dSYM compiled from bitcode to the original
+/// UUID of the matching [`BCSymbolMap`].
+#[derive(Clone, Debug)]
+pub struct PList {
+    id: DebugId,
+    map: HashMap<String, String>,
```

Comment:
Since there's little we can do with `PList`s other than use them for dSYMs, we can simply read the UUID and other relevant attributes, but discard the rest. Particularly, this allows us to provide an infallible signature for `PList::original_uuid`.